homeassistant/components/modbus/climate.py

```diff
@@ -255,7 +255,15 @@ def _read_register(self, register_type, register) -> Optional[float]:
         byte_string = b"".join(
             [x.to_bytes(2, byteorder="big") for x in result.registers]
         )
-        val = struct.unpack(self._structure, byte_string)[0]
+        val = struct.unpack(self._structure, byte_string)
+        if len(val) > 1 or not isinstance(val[0], (float, int)):
```

Comment:
Can the length ever be 0?

```suggestion
        if len(val) != 1 or not isinstance(val[0], (float, int)):
```

Reply:
In case of error, field result.registers is not defined and consequently byte_string would fail before.
A request with count == 0 that would result in empty result.registers is not allowed, fails with exception.

Reply:
That is not exactly true, you cannot configure 0 registers, but the device can very well decide not to deliver any registers.

If the device does not deliver any registers (e.g. because the start register + count goes into an invalid range) then we have the 0 situation as a consequence testing for > 1 makes sense to me.

Reply:
> Can the length ever be 0?

I don't know, but your suggestion looks OK to me. I will apply that.